### PR TITLE
PRESIDE-1056 Allow for sites to run in subfolder of root

### PR DESCRIPTION
**Pull request by patmoody**
Application config example
settings.static.rootUrl = "/cms_p";
settings.preside_admin_base_path = "cms_p/";

Application.cfc
super.setupApplication( id = "mysite", appBasePath = "/cms_p" );

Site path in DB
update psys_site set path = '/cms_p';

### CODE REVIEW

**Comment by patmoody** [comment]
This is a start but still needs some work! Failing in a standard install with site in root location. Looks like some Asset management pathing issues too with site outside root.

**Comment by patmoody** [comment]
Fixed for standard installation. Asset management pathing still a problem for non standard (Site in subfolder)

**Comment by patmoody** [comment]
Fixed for assets too. 
@DominicWatson 
Ready for review.

File: system/services/assetManager/AssetManagerService.cfc [hunk 1]
```diff
@@ -1870,6 +1874,14 @@ component displayName="AssetManager Service" {
 		_setTypes( types );
 	}
 
+	private string function _getAssetPath( string appBasePath="/asset/" ) {
```
Comment (patmoody) [inline system/services/assetManager/AssetManagerService.cfc]: This of course should be _getAssetPath() - Will await other instructions before committing again

**Comment by DominicWatson** [comment]
Fab, thanks for this @patmoody. I'll take a good look shortly.

File: system/Bootstrap.cfc [hunk 2]
```diff
@@ -546,6 +551,32 @@ component {
 		return ExpandPath( "/preside" );
 	}
 
+	private string function _getAppBasePath( string appBasePath ) {
+		var appBasePath = "";
+
+		if ( Len( arguments.appBasePath ) ) {
+			appBasePath = arguments.appBasePath;
+		} else {
+			var cbController = _getColdboxController();
```
Comment (patmoody) [inline system/Bootstrap.cfc]: @DominicWatson 
Trying to get this to work based on site.path if no configuration options are passed through / set in the applications config.
Thought I had this working locally but just realised I was passing the site path in the applications app cfc file.
cbController is returning as null when cfg options are removed.

**Comment by patmoody** [comment]
Latest commit was trying to avoid settings in the applications config and application.cfc and basing the root path on the site.path. - bootstrap.cfc - _getAppBasePath
cbController = _getColdboxController() 
returns null

**Comment by DominicWatson** [comment]
Hey Pat, apologies I've not managed to look at this yet. I'll defo look at this before we release 10.9. Thanks for keeping going with it

**Comment by patmoody** [comment]
No probs. My problem now is checking the site.path programmatically from the bootstrap.cfc ^^

**Comment by DominicWatson** [comment]
Right, finally getting round to looking into this and there appears to be an issue with the pull request. The branch you target the pull request back into should be the branch you have based your feature branch from. e.g. if you did:

```git checkout -b mynewfeature stable```

or, you had `stable` checked out and you did:

```git checkout -b mynewfeature```

...then your pull request should be aimed at `stable`. There are a lot of changes here unrelated to your work I presume (160+ files changed). 

Could you take a look and resubmit by closing this and submitting a new pull request? See our contrib guidelines for more details: https://beta-docs.preside.org/contribguides/submittingchanges.html

**Pull request closed without merging**